### ISSUE TITLE
Protege distribution changes break for all but bundled versions of jre

### ISSUE THREAD

**Issue by sesuncedu** [protegeproject/protege#163]
Because OSGI.  Also, the bundled jre has a really odd L&F on OS-X (not aqualine).

The platform-independent version doesn't work at all. 
Investigations continue.  I continue to have absolute confidence in OSGI.

```
+ java -Xmx500M -Xms250M -Dlog4j.configuration=file:log4j.xml -DentityExpansionLimit=100000000 -Dfile.encoding=UTF-8 -classpath bin/org.apache.felix.main.jar:bin/ProtegeLauncher.jar org.protege.osgi.framework.Launcher
ERROR: Bundle org.protege.common [1] Error starting file:/Users/ses/src/Semantic/protege/protege-distribution/target/protege-5.0.0-beta-18-SNAPSHOT-platform-independent/Protege-5.0.0-beta-18-SNAPSHOT/bundles/org.protege.common.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle org.protege.common [1]: Unable to resolve 1.0: missing requirement [1.0] osgi.wiring.package; (osgi.wiring.package=javax.management))
org.osgi.framework.BundleException: Unresolved constraint in bundle org.protege.common [1]: Unable to resolve 1.0: missing requirement [1.0] osgi.wiring.package; (osgi.wiring.package=javax.management)
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4002)
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2045)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1299)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:304)
	at java.lang.Thread.run(Thread.java:745)
```
And many, many more.

**Comment by sesuncedu** [comment]
Further clue: 
In intelliJ, unless I move bin/org.felix.main.jar to the head of the class path, things break.
Still not getting apple L&F.

**Comment by sesuncedu** [comment]
The macos  detection code hasn't worked for a long long time;  looking for system containing "os x", instead of "macos". 

Still hasn't made OS.X L&F reappear.

**Comment by sesuncedu** [comment]
Really really really doesn't want to work with anything but a bundled jre.

**Comment by sesuncedu** [comment]
The check box is from aqua L&F.  
The tabs are from somewhere else.
![part-mac-part-not](https://cloud.githubusercontent.com/assets/411675/6723121/8748e33e-cdb9-11e4-86f1-42e457722be8.png)

**Comment by sesuncedu** [comment]
In fact, Tabs seem to be the major problem;  JTree switches between metal and plastic pretty well.
Also, L&F changes dynamically, so the dialog box that shows up telling you that l&f will change on restart is not only lying, it's rendered using the l&f you just selected.

**Comment by sesuncedu** [comment]
Also: http://wiki.osgi.org/wiki/DynamicImport-Package

**Comment by sesuncedu** [comment]
Revert, or advance?

**Comment by sesuncedu** [comment]
Also, I had intellij working enough so that I could have it generate fresh bundles, and use the debugger on loaded classes.  Now I don't.   
@matthewhorridge , have you gotten intellij debugging working with the new packaging stuff?

Also also,  some dependency or another is compiled using java 1.5 byte code.  How do I know?  Bundle resolution fails with no useful errors if jdk 1.9 gets picked up (I accidentally did a make install instead of make images earlier this week, and had a 'java' in /usr/local/bin which was getting picked up instead of going via JAVA_HOME )

**Comment by matthewhorridge** [comment]
The tabs are now custom and are actually the same across all look and feels.  While they don't look like the old aqua tabs (apple has moved on from these on Yosemite anyway) they are more compact and they are also closable.  It's possible that they will evolve over time.

With regards to switching look and feels, parts of the look and feel change and other parts don't get updated so well, so a restart is always advisable.

For debugging in IntelliJ (and this is cumbersome) I run maven package and then start Protege with the launcher and the working directory specified as the target.  It's kind of painful, but I've been doing this for a long time since Tim switched over to OSGi - everything seemed too flaky otherwise.  I'd be interested in improving this for sure.

**Comment by sesuncedu** [comment]
I had been debugging like that, but not all breakpoints seem to be being
hit ( class path hell?) I did have the facets configured so that I could
make a quick change and have intellij build the bundle from the OSGI facet
(after having done a maven build once to generate the manifest).

Maven building isn't so annoying since I found the reason why the OWL
editor always recompiles everything (two classes don't match their files).

I have a bunch of pom related changes to cherry-pick from.
owlapi 3.5.2 is needed for OBOing, and protege-owlapi could do with going
to 3.2.7. There have been  too many non-backwards-compatible api changes in
the 3.5 series, and too many bits have compiled against different versions,
in ways that are not guarded by OSGI. I think animal sniffer baseline
comparisons may be more use.

I have a deepening need to find some place where OSGI is actually helpful
(I think it might have allowed to bundles with declared dependencies on
different Guava versions to be installed, but since both versions were
effectively compatible, it's more a failure to lose than an actual win).

Maybe if there were pure api bundles it would work.  Or if everything was a
set top box.
